### DISCUSSION TITLE
Docker install failing

### DISCUSSION THREAD

**Discussion by firstfilterservice** [opensourcepos/opensourcepos#3795] [Q&A] [answered]
Hello!

I'm trying to install ospos via Docker. The system is Proxmox -> Debian 11 -> Docker with Portainer. I am totally new to Docker (and Debian and Proxmox), so please explain it like I'm 5! 

I tried the Dockerfile in Portainer without success. Then I tried via CLI and both fail in the same spot with the same error message. I wasn't sure if Portainer was the problem, but it doesn't seem to be.

The error message is:

> Step 10/24 : RUN chmod -R 750 /app/public/uploads /app/application/logs && chown -R www-data:www-data /app/public /app/application
> 
> 
> 
>  ---> Running in b9401b058462
> 
> chmod: cannot access '/app/public/uploads': No such file or directory
> chmod: cannot access '/app/application/logs': No such file or directory
> 
> The command '/bin/sh -c chmod -R 750 /app/public/uploads /app/application/logs && chown -R www-data:www-data /app/public /app/application' returned a non-zero code: 1

Any ideas as to what I'm doing wrong?

Thanks!

**Comment by jekkos** [comment]
If you use proxmox isn't it 

Debain -> Proxmox -> Debian -> Docker with Portainer?

It seems that you might be missing some directories for some reason. I would start from the docker compose instead. I have not worked with portainer so not sure how to run it through their UI.

**Comment by firstfilterservice** [comment]
Yes, you are quite correct - Debain -> Proxmox -> Debian -> Docker with Portainer. I'm wondering if something went wrong installing Docker so tonight I installed a Docker LCX via a script (a potentially unwise idea but hey, gotta learn somehow!) It seemed to go much smoother.

Installing with the Dockerfile in Portainer resulted in the same error however. I then tried to pull an image from Docker Hub and that seems to have worked well. I cannot access anything on the ip of the container as it's set to something strange (172.x.x.x when my local network is 192.168.1.x), but I will continue to work on that. If I cannot get the IP address figured out, I will take a run at Docker compose and see if that will let me set an IP for the container.

Thanks!

**Reply by jekkos** [reply]
The 172.x ip is the internal docker network, you need to expose the port to the host. This is already done in the docker compose.

**Reply by firstfilterservice** [reply]
Ok, I'm making progress! I now get an error from the webserver rather than nothing at all. However, I think I'm going to take your advice and  try docker compose.

**Comment by firstfilterservice** [comment]
More progress. I've read up on Docker compose and Docker in general and I have found my main issue. It might help total Linux noobies like myself if the Docker install instructions looked more like this:

1. wget opensourcepos.3.3.8.ffe492.zip
2. unzip opensourcepos.3.3.8.ffe492.zip
3. docker compose up

I had to learn how to download files from the web with the CLI. Then I downloaded the Dockerfile thinking it was going to download everything it needed, but I really needed to download the project zip file and start with that. 

That all said - I'm still not at a working install. My docker compose up seems to be frozen at:
![docker compose up stuck](https://github.com/opensourcepos/opensourcepos/assets/37482898/1174a322-91b8-46aa-a61b-83b403f58777)

It's been there a solid 10 minutes. I've canceled (Control - C) the install and re-run docker compose up to see if it was a glitch, but it stops in exactly the same place. 

I went back to the Getting Started wiki page on install, it mentioned to run docker compose build before docker compose up, so I canceled and ran the build, then re-ran up and still stuck in the same spot.

So I started looking at docker compose vs docker-compose. Went down a little rabbit hole. Installed docker-compose using a guide. Failed. Found another guide (https://cloudinfrastructureservices.co.uk/how-to-install-and-use-docker-compose-on-debian-11/) that worked and now docker-compose is installed so I'm golden right? Nope. Stuck in the same place!

So that's where things are at for today. I'll continue to work on this as I feel like I'm making progress. I'm documenting this all here for future generations. Any suggestions will be greatly appreciated, thank you!

**Reply by firstfilterservice** [reply]
Ok, I couldn't quit - there's small sentence in the wiki "Download the latest version of from the [release section](https://github.com/opensourcepos/opensourcepos/releases) and uncompress it to the htdoc directory" - where is the htdoc directory? I just unzipped the file in my /root/osposdocker folder and docker-compose-ed is there. Perhaps that's where I've gone astray. I can't seem to find a htdoc folder anywhere but I'll keep looking.

**Reply by jekkos** [reply]
Hi with htdoc it probably means the web server document root. I don't think it's relevant in case of docker compose. It should just work without webserver.

**Reply by firstfilterservice** [reply]
I haven't been able to get the docker compose to work. I gave up on the docker lxc and spun up a few new dedicated vm's to test with, that way I can rule out the lxc as a possible source of problems. 

Following the wiki Getting Started Local Docker instructions = Docker compose on a fresh Debian 11 vm looked like it completed, but no access from a web browser. But I was trying to run the compose from the "htdoc" folder (/var/www/html) so I think that might have tripped me up. I'll try from the "osposdocker" directory tomorrow and see what that gets me. I'll probably spin up a new vm just to be fresh.

I did have some success following this guide: https://www.opensourceposguide.com/guide/gettingstarted/installation and installing things manually. I actually got a login screen! But once I put in the username and password I get a "Not found. The requested URL was not found on this server" error. I'm thinking the problem is the htaccess as the guide has "(Note: Make sure you have set up .htaccess files for the html directory. DigitalOcean does this for you automatically.)" And I haven't been able to make heads or tails from that.

Thanks!

**Comment by firstfilterservice** [comment, marked as answer]
Success! I went back to the Docker compose (aka Portainer Stack) and inspected the network section carefully looking for differences between it and the Portainer container. I found that the "Port configuration" in the working Portainer container was "0.0.0.0:8000:8000" while the ospos port configuration was "127.0.0.1:80:80". I edited the ospos port configuration to "0.0.0.0:80:80" and re-started the stack and ospos came up immediately in my browser. 

Once I get a bit more familiar I will write up a quick Portainer specific guide and post it here. I think Portainer is a bit more accessible to noobies like me.
